### DISCUSSION TITLE
Allow bubbletea programs to set input/output.

### DISCUSSION THREAD

**Discussion by n3wscott** [charmbracelet/bubbletea#335] [General]
See https://github.com/charmbracelet/bubbletea/blob/master/tea.go#L257-L258

There does not seem to be an option to set input/output to something other than stdin/stdout. It would be better go-ism to let the program struct export those member fields rather than protect them and use the option methods unless you have a very good reason, or you need full options to set those inner fields.

As the `Program` struct sits today, it is not very unit testable.

**Comment by muesli** [comment]
You can initialize a new `tea.Program` with `WithInput` and `WithOutput` as functional options: https://github.com/charmbracelet/bubbletea/blob/master/options.go#L16

There's also some more work in progress in this branch: https://github.com/charmbracelet/bubbletea/tree/termenv-output
